### PR TITLE
[bugfix]: Adds meta descriptions to root pages

### PR DESCRIPTION
**Pull request by sirugh**
## Description

This uses the `Meta` component from `react-head` to provide the description meta tag.

I updated all the queries to return it regardless of whether we're on server or not. I then configured each component to utilize the returned value in the rendered HTML.

If lighthouse still complains with this change that means the sample data/backend data needs to be updated. 

[Image from Gyazo](https://gyazo.com/4fb9c7afc48280f71a2b28a2b0130dc9)

I'm not sure if JS crawlers wait for/allow this sort of async update of meta description. We could possibly provide a default value?

## Related Issue

Closes [PWA-217](https://jira.corp.magento.com/browse/PWA-217).

### Verification Steps

1. Go to the app, check the HTML for a meta description tag.
2. Navigate to the category and check that it is still there.
3. Do the same for a product.

You may notice that `content` is empty for some pages. If that happens it is because the backend is returning `null`. Check the response for the relevant query to see if that is true.

## Checklist

* I have updated the documentation accordingly, if necessary.
* I have added tests to cover my changes, if necessary.

### CODE REVIEW

File: packages/venia-ui/lib/RootComponents/CMS/cms.js [hunk 1]
```diff
@@ -36,7 +37,15 @@ const CMSPage = props => {
             return <RichContent html={data.cmsPage.content} />;
         }
 
-        return <CategoryList title="Shop by category" id={2} />;
+        return (
+            <>
+                <Meta
+                    name="description"
+                    content={data.cmsPage.meta_description}
```
Comment (sirugh) [inline packages/venia-ui/lib/RootComponents/CMS/cms.js]: @davemacaulay do you guys want this in the above conditional?

Comment (davemacaulay) [reply packages/venia-ui/lib/RootComponents/CMS/cms.js]: It makes sense to me to have it for all resolved pages, the above conditional was to ensure the category list shows up on instances without Page Builder.

Comment (sirugh) [reply packages/venia-ui/lib/RootComponents/CMS/cms.js]: Right -- I'm considering just adding this to all pages but this specific changeset was supposed to address the lighthouse score dinging us for not having the description on the root page (which happens to be CMS).

Comment (davemacaulay) [reply packages/venia-ui/lib/RootComponents/CMS/cms.js]: It should then be included on all CMS pages, correcting meta data for all pages is more beneficial then a lighthouse score fix imo.

Comment (sirugh) [reply packages/venia-ui/lib/RootComponents/CMS/cms.js]: Agreed :) I went ahead and added it to everything.

File: packages/venia-ui/lib/queries/getProductDetailBySku.graphql [hunk 2]
```diff
@@ -38,9 +38,6 @@ query productDetailBySku($sku: String, $onServer: Boolean!) {
                     }
                 }
             }
-            meta_title @include(if: $onServer)
-            meta_keyword @include(if: $onServer)
-            meta_description @include(if: $onServer)
```
Comment (sirugh) [inline packages/venia-ui/lib/queries/getProductDetailBySku.graphql]: These were unused. This query is only used for the cart and shouldn't need these details anyways.

File: packages/venia-ui/lib/fragments/productDetails.graphql [hunk 3]
```diff
@@ -21,7 +21,7 @@ fragment ProductDetails on ProductInterface {
     }
     meta_title @include(if: $onServer)
     meta_keyword @include(if: $onServer)
-    meta_description @include(if: $onServer)
```
Comment (sirugh) [inline packages/venia-ui/lib/fragments/productDetails.graphql]: We now want `meta_description` regardless of whether we are doing SSR. I assume there is still a path to use this in SSR as there are templates in our repo that use these meta fields but I couldn't figure out how exactly.

**Comment by sirugh** [comment]
> I'm also curious what the {category, cmspage, product}-shell.mst templates are there for, which appear to fulfill this requirement; but not a blocker.

I think they are related to SSR. @zetlen may know?

**Pull request merged**